libraries/libutils/BinaryWriter.h

```diff
@@ -16,7 +16,7 @@ class BinaryWriter
         put_data(bytes, sizeof(T));
     }
 
-    inline void put_data(uint8_t *data, unsigned int len)
+    inline void put_data(uint8_t *data, size_t len)
     {
         _data.push_back_many({ADOPT, data, len});
```

Comment:
This look sketchy, the object will adopt the memory by we don't know if it can be free'd